qui/devices/device_widget.py

```diff
@@ -124,7 +124,8 @@ def device_list_update(self, vm, _event, **_kwargs):
         try:
             for devclass in DEV_TYPES:
                 for device in vm.devices[devclass]:
-                    changed_devices[str(device)] = backend.Device(device, self)
+                    changed_devices[str(device.port)] = backend.Device(
```

Comment:
if the variable changed_devices is now a dict that maps port to device, that should be reflected in the name or at least in comments